terraform/modules/tf_stream_alert/iam.tf

```diff
@@ -42,6 +42,29 @@ data "aws_iam_policy_document" "rule_processor_invoke_alert_proc" {
   }
 }
 
+// IAM Role Policy: Allow the Rule Processor to publish CloudWatch metrics
+resource "aws_iam_role_policy" "streamalert_rule_processor_cloudwatch_put_metric_data" {
```

Comment:
Instead of creating a separate policy for metrics, all of the Lambda functions could inherit the same base policy with permission to publish logs and metrics (e.g. https://github.com/airbnb/binaryalert/blob/master/terraform/lambda_iam.tf#L3-L17)